internal/providers/azurestack/azurestack.go

```diff
@@ -34,5 +34,5 @@ const (
 
 // FetchConfig implements the platform.NewFetcher interface.
 func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
-	return azure.FetchFromOvfDevice(f, []string{CDS_FSTYPE_UDF, CDS_FSTYPE_ISO9660})
+	return azure.FetchFromAzureMetadata(f, []string{CDS_FSTYPE_UDF, CDS_FSTYPE_ISO9660})
```

Comment:
Have you confirmed that Azure Stack always supports the metadata service?  If not, we should probably keep `FetchFromOvfDevice` here, and make `FetchFromAzureMetadata` a wrapper around it.

Reply:
Right - thanks for the heads-up ! According to the documentation[^1]:

> The Azure Instance Metadata Service isn't supported on Azure Stack Hub.

[^1]: https://docs.microsoft.com/en-us/azure-stack/user/azure-stack-vm-considerations